stackoverflow/src/main/java/ro/utcn/sd/cata/stackoverflow/controller/ConsoleController.java

```diff
@@ -0,0 +1,227 @@
+            case "addAnswer":
+                handleAddAnswer();
+                return false;
+            case "editAnswer":
+                handleEditAnswer();
+                return false;
+            case "removeAnswer":
+                handleRemoveAnswer();
+                return false;
+            case "exit":
+                return true;
+            default:
+                print("Unknown command. Try again.");
+                return false;
+        }
+    }
+
+    private void handleAddUser(){
+        print("Username:");
+        String username = scanner.next().trim();
+        print("Password:");
+        String password = scanner.next().trim();
+        User user = userService.addUser(username, password);
+        print("Created user: " + user + ".");
+    }
+
+    private void handleRemoveUser(){
+        print("User id:");
+        Integer id = scanner.nextInt();
+        userService.removeUser(id);
+        print("User removed.");
+    }
+
+    private void handleLogin(){
+        print("Enter username:");
+        String username = scanner.next().trim();
+        print("Enter password:");
+        String password = scanner.next().trim();
+        Optional<User> user = userService.findUserByUsernamePassword(username,password);
+        if(user.equals(Optional.empty())){
+            print("User or Password incorrect. Please try again");
+        }
+        else{
+            loggedUser =  user.get();
+            loggedIn = true;
+            print("Login successful. Welcome "+loggedUser.getUsername()+"! \n");
+        }
+    }
+
+    private void handleAddQuestion(){
+        if(loggedIn){
+            String getNewlineRemainedFromPreviousCommand = scanner.nextLine().trim();
+            print("Enter title:");
+            String title = scanner.nextLine().trim();
+            print("Enter text:");
+            String text = scanner.nextLine().trim();
+            Question addedQuestion = questionService.addQuestion(title,text,loggedUser);
+            boolean doneAddingTags = false;
+            while(!doneAddingTags){
+                print("Enter tag or type done: ");
+                String tag = scanner.next().trim();
+                if(tag.equals("done")){
+                    doneAddingTags=true;}
+                else{
+                    tagService.addTag(tag,addedQuestion);}
+            }
+            print(("Question added: "+ addedQuestion.toString()));
+        }
+        else print("Log in in order to add questions.");
+    }
+
+    private void handleList(){
+        if(loggedIn){
+            for(Question question: questionService.findAllOrderByDate())
+                print(question.toString());
+        }
+        else print("Log in in order to see questions.");
+    }
+
+    private void handleFindTag(){
+        if(loggedIn){
+            print("Write the tag: ");
+            String tag = scanner.next().trim();
+            if(tagService.findByName(tag).isPresent()){
```

Comment:
See: here you used the `isPresent` ;) 

Also, it would have been an idea to always return a tag from the service layer and throw an exception if not found. As discussed during the lab, exception handling in the controller layer will anyway be needed later.